stonesoup/hypothesiser/distance.py

```diff
@@ -53,3 +54,72 @@ def hypothesise(self, track, detections, timestamp):
             self.missed_distance))
 
         return sorted(hypotheses, reverse=True)
+
+
+class GMMahalanobisDistanceHypothesiser(Hypothesiser):
```

Comment:
So looking at the implementation, you should be able to create a `GaussianMixtureHypothesier`, which simply passes each component of the mixture to the existing hypothesiser implementations. This will make this more generic, and able to work with other hypothesisers in future.

Reply:
For the PHD/LCC filters its best if the hypotheses are collected by measurement/detection rather than by component since we need to normalise the weights per measurement. So in order to use other hypothesiers, there might need to be a bit of code afterwards in the `GaussianMixtureHypothesier` that rearranges the list of `MultipleHypothesis` to suit. Would that be a suitable approach?

Reply:
Yes, that would work.